lib/listext.ml

```diff
@@ -229,4 +229,12 @@ let range lower =
 		else aux (upper-1::accu) (upper-1) in
 	aux []
 
+let rec assoc_opt x = function
+| [] -> None
+| (a,b)::l -> if compare a x = 0 then Some b else assoc_opt x l
```

Comment:
Is `compare` available in the scope? The other usage in the module of the compare function it gets introduced as a function parameter.

Reply:
`compare` should be in scope (in 408 it is at least)

```
[ben@piglet:~ {master408}]$ utop
────────┬─────────────────────────────────────────────────────────────┬─────────
        │ Welcome to utop version 2.4.2 (using OCaml version 4.08.1)! │         
        └─────────────────────────────────────────────────────────────┘         

Type #utop_help for help about using utop.

─( 11:41:19 )─< command 0 >──────────────────────────────────────{ counter: 0 }─
utop # compare;;
- : 'a -> 'a -> int = <fun>
─( 11:41:19 )─< command 1 >──────────────────────────────────────{ counter: 0 }─
utop # 

```

i think it is passed as a param to other functions to give the caller more flexibility

Reply:
compare is part of Pervasives.